crates/ruff_python_formatter/src/context.rs

```diff
@@ -1,11 +1,22 @@
 use crate::comments::Comments;
-use crate::string::QuoteChar;
+use crate::string::{QuoteChar, StringQuotes};
 use crate::PyFormatOptions;
 use ruff_formatter::{Buffer, FormatContext, GroupId, IndentWidth, SourceCode};
 use ruff_source_file::Locator;
 use std::fmt::{Debug, Formatter};
 use std::ops::{Deref, DerefMut};
 
+/// Location of the expression which is currently being formatted.
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) enum ExpressionLocation {
+    /// The expression is inside an f-string, in the replacement field i.e., `f"foo {x}"`.
+    ///
+    /// The containing `StringQuotes` is surrounding f-string quote information.
+    InsideFString(StringQuotes),
+    #[default]
+    Other,
```

Comment:
Nit: I would rename the enum to `FStringContext` or similar and have a variant `Inside` or `Outside` because the enum doesn't represent arbitrary expression contexts.

Reply:
`FStringContext` is already taken 😅

Let me think of something else.